AjaxControlToolkit/AjaxFileUpload/AjaxFileUpload.cs

```diff
@@ -389,11 +389,20 @@ public AjaxFileUpload()
         }
 
         public static string BuildRootTempFolder() {
-            var rootTempFolder = Path.Combine(AjaxFileUploadHelper.RootTempFolderPath, TemporaryUploadFolderName);
-            if(!Directory.Exists(rootTempFolder))
-                Directory.CreateDirectory(rootTempFolder);
+            var userPath = AjaxFileUploadHelper.RootTempFolderPath;
 
-            return rootTempFolder;
+            if(userPath != null) {
+                if(!Directory.Exists(userPath))
+                    throw new IOException(String.Format("Temp directory '{0}' does not exist."));
+                return userPath;
+            }
+
+            var defaultPath = Path.Combine(Path.GetTempPath(), TemporaryUploadFolderName);
```

Comment:
For better readability, let's rename `TemporaryUploadFolderName` const to `DefaultTempSubDir`, or something similar